### PR TITLE
chore(FDS-255): Add input date field

### PR DESCRIPTION
**Pull request by rociobaena**
## Input date field
The input date type is adding to our module list.
This PR includes:

- Module DataDate.
- Fixture in AllModules.fixture.js file.
- Unit test.

- [x] Includes unit tests for _ALL_ required FDS use cases
- [x] Does not mute any top level API props in React
- [x] Includes an explanation for any changes to Jest snapshots (should be a PR tag automatically added if this happens)
- [-] Component uses `React.forwardRef()` when a single DOM node is returned, _AND_ it makes React-sense to access that DOM node

### Snapshots
A new fixture for the DataDate field is created.

### CODE REVIEW

File: packages/cascara/src/modules/DataDate/index.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export { default, propTypes } from './DataDate';
```
Comment (brianespinosa) [inline packages/cascara/src/modules/DataDate/index.js:1]: Are we actually using the prop types for this module outside of the module itself?

**Review by brianespinosa — CHANGES_REQUESTED** [review]
@rociobaena Just a couple small things. Mainly that I think it would make sense to create a fixture file for each module which we export fixtures from to write tests against.

Your first module! 👏🏽

**Review by lordpixel — APPROVED** [review]

File: packages/cascara/src/modules/DataDate/DataDate.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    <label htmlFor={label}>
+      {label && isLabeled && <span className={styles.LabelText}>{label}</span>}
+      <Input
+        {...rest}
+        aria-label={label}
+        className={styles.Input}
+        defaultValue={value}
+        id={label}
+        max={max}
+        min={min}
```
[the remark this answers is not in the document]
Comment (rociobaena) [reply packages/cascara/src/modules/DataDate/DataDate.js]: Properties removed, and rely on ...rest instead.

File: packages/cascara/src/modules/DataDate/DataDate.test.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+  describe('editing', () => {
+    // specifying ModuleSandbox will have the flag to render our input date field
+    const testId = 'default';
+
+    beforeEach(() =>
+      render(
+        <ModuleSandbox isEditing>
+          <DateDate data-testid={testId} label='Date' value={VALUE} />
+        </ModuleSandbox>
```
[the remark this answers is not in the document]
Comment (rociobaena) [reply packages/cascara/src/modules/DataDate/DataDate.test.js]: @brianespinosa I tried to do what you said, take a look and let me know what you think.

File: packages/cascara/src/modules/DataDate/index.js (hunk 1 above)
Comment (rociobaena) [reply packages/cascara/src/modules/DataDate/index.js]: propTypes is not being used outside of this module. Code removed.

**Review by lordpixel — APPROVED** [review]

File: packages/cascara/src/modules/DataDate/DataDate.fixture.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        <p>
+          This fixture is for testing the presentation of an input date field
+        </p>
+      </div>
+
+      <hr />
+
+      <div>
+        <h2>Display</h2>
+        <DisplayDataDate isEditing={false} value={'2021-07-24'} />
```
Comment (brianespinosa) [inline packages/cascara/src/modules/DataDate/DataDate.fixture.js:31]: If we have a boolean prop, we do not need to define false on it. We can safely leave off isEditing to set it as false.

File: packages/cascara/src/modules/DataDate/DataDate.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  value: pt.string,
+};
+
+const DataDate = ({
+  attribute,
+  isEditable = true,
+  isLabeled = true,
+  label,
+  max,
+  min,
```
Comment (brianespinosa) [inline packages/cascara/src/modules/DataDate/DataDate.js:30]: We removed `min` and `max` properties below on the input... but we are still destructuring them here on the DataDate function. That is effectively muting them from being passed to the Input component with `{...rest}`. We should remove them here so we do not prevent them from passing down.

File: packages/cascara/src/modules/DataDate/DataDate.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  /** A module can have an Attribute, which will be used as form field name */
+  attribute: pt.string,
+  /** A Module can be defined to not present an editing state */
+  isEditable: pt.bool,
+  /** Presents the input without a label. NOT USER CONFIGURABLE */
+  isLabeled: pt.bool,
+  /** A Module needs to have a unique label relative to its context */
+  label: pt.string,
+  max: pt.string,
+  min: pt.string,
```
Comment (brianespinosa) [inline packages/cascara/src/modules/DataDate/DataDate.js:19]: We should also remove these props from our prop-types if we are just passing them with `...rest`

File: packages/cascara/src/modules/DataDate/DataDate.test.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+      expect(input.tagName).toMatch('SPAN');
+    });
+  });
+
+  describe('editing', () => {
+    // specifying ModuleSandbox will have the flag to render our input date field
+    const testId = 'default';
+    beforeEach(() =>
+      render(
+        <DisplayDataDate data-testid={testId} isEditing value={'2021-07-24'} />
```
Comment (brianespinosa) [inline packages/cascara/src/modules/DataDate/DataDate.test.js:39]: 🙌🏽

**Review by brianespinosa — CHANGES_REQUESTED** [review]
@rociobaena close. A couple of those min/max props are still in our component, and we can remove the isEditing={false} in the fixture. We are ready to merge after those few changes!

**Review by brianespinosa — APPROVED** [review]

**Comment by brianespinosa** [comment]
@rociobaena I approved but it looks like we still have a test failing. I suspect it is for a snapshot test not getting updated with the new module name.

**Review by lordpixel — DISMISSED** [review]
👍

**Comment by lordpixel** [comment]
@rociobaena I think we need to update snapshots

**Review by lordpixel — APPROVED** [review]
👍

**Review by brianespinosa — APPROVED** [review]

**Pull request merged**